plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -427,15 +426,16 @@ func podName(pod *api.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
-func PodFitsResources(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+func PodFitsResources(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
 	node := nodeInfo.Node()
 	if node == nil {
-		return false, fmt.Errorf("node not found")
+		return false, nil, fmt.Errorf("node not found")
 	}
+
+	predicateFails := make([]algorithm.PredicateFailureReason, 0)
```

Comment:
var predicateFails []algorithm.PredicateFailureReason

should also work and would avoid memory allocation if this is not needed.